src/game/CreatureEventAI.cpp

```diff
@@ -1366,11 +1366,18 @@ void CreatureEventAI::UpdateAI(const uint32 diff)
         // Update creature dynamic movement position before doing anything else
         if (m_DynamicMovement)
         {
-            if (!m_creature->hasUnitState(UNIT_STAT_CAN_NOT_REACT) && !m_creature->IsNonMeleeSpellCasted(false))
+            if (!m_creature->hasUnitState(UNIT_STAT_CAN_NOT_REACT))
             {
-                if (m_LastSpellMaxRange && m_creature->IsInRange(m_creature->getVictim(), 0, (m_LastSpellMaxRange / 1.5f)))
-                    SetCombatMovement(false, true);
-                else
+                bool uiInLineOfSight = m_creature->IsWithinLOSInMap(m_creature->getVictim());
```

Comment:
This variable can be moved inside the if statement on line 1373 - I would also move the bool check to be the first instead of the spellcast check as that would seem heavier to me.